test/integration/QtiPackageExportTest.php

```diff
@@ -0,0 +1,116 @@
+use Slim\Http\Request;
+use Slim\Http\RequestBody;
+use Slim\Http\Uri;
+use tao_helpers_File;
+use taoQtiTest_actions_RestQtiTests;
+use Zend\ServiceManager\ServiceLocatorInterface;
+use common_report_Report as Report;
+
+class QtiPackageExportTest extends RestTestRunner
+{
+    private const GET_REQUEST = 'GET';
+    /** @var ServiceLocatorInterface */
+    private $serviceLocatorMock;
+
+    public function setUp()
+    {
+        parent::setUp();
+        $this->serviceLocatorMock = $this->getServiceLocatorMock([
+            QtiPackageExporter::class => new QtiPackageExporter(),
+        ]);
+    }
+
+    public function testExportQtiPackage()
+    {
+        $exportTempFile = tempnam(sys_get_temp_dir(), 'testExport_');
+
+        //create test resource based on qti package zip
+        $testFile = __DIR__ . '/samples/archives/QTI 2.2/exportWithoutLongPaths/multiple_items_with_ms.zip';
+        //create temporary subclass
+        $class = \taoTests_models_classes_TestsService::singleton()->getRootclass()->createSubClass(uniqid('test-exporter', true));
+        //Importing test form zip file into temporary subclass
+        /** @var Report $report */
+        $report = \taoQtiTest_models_classes_QtiTestService::singleton()
+            ->importMultipleTests($class, $testFile);
+        $reportSuccesses = $report->getSuccesses();
+        $successfulImport = reset($reportSuccesses);
+        $this->assertCount(1, $reportSuccesses);
+        $this->assertEquals($report->getType(), \common_report_Report::TYPE_SUCCESS);
+        $this->assertFalse($report->containsError());
+
+        //Detremine what uri our new class was defined in tao
+        $uriResource = $successfulImport->getData()->uriResource;
+        //Encode resource uri so it can be passed to url param
+        $testUriEncoded = urlencode($uriResource);
+        $query = sprintf('?testUri=%s', $testUriEncoded);
+
+        $restQtiTests = new TestableRestQtiTests();
+        $restQtiTests->setServiceLocator($this->serviceLocatorMock);
+
+        //Preparing right url
+        $stream = new RequestBody();
+        $headers = new Headers();
+        $request = new Request(self::GET_REQUEST, Uri::createFromString($query), $headers, [], [], $stream);
+        $restQtiTests->setRequest($request);
+
+        //Execute
+        $response = $restQtiTests->exportQtiPackage();
+
+        //Handle Zip file
+        $filehandler = fopen($exportTempFile, 'wb');
+        fwrite($filehandler, base64_decode($response[TestableRestQtiTests::PARAM_PACKAGE_NAME]));
+        fclose($filehandler);
+
+        $extractExportPath = sys_get_temp_dir() . '/extractedExportPackage';
+        $zip = new \ZipArchive();
+        if ($zip->open($exportTempFile)) {
+            $zip->extractTo($extractExportPath);
+            $zip->close();
+        }
+        $manifestFile = $extractExportPath . '/imsmanifest.xml';
+
+        //Assert zip content
+        $this->assertTrue(file_exists($manifestFile));
```

Comment:
Would it be safer to check that all the required file exist?

Reply:
We need to check if the export process was successful and zip file can be decompressed. This is the reason that I check if extracted files contain expected manifest file. The additional check would test if exporter working correctly which I think out of the scope of this test.